examples/access-control-migration/src/services/assign-project-instance-id.voter.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-access-control-migration
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  AuthorizationContext,
+  AuthorizationDecision,
+  AuthorizationMetadata,
+} from '@loopback/authorization';
+
+/**
+ * Instance level authorizer for known endpoints
+ * - 'projects/{id}/show-balance'
+ * - 'projects/{id}/donate'
+ * - 'projects/{id}/withdraw'
+ * This function is used to modify the authorization context.
+ * It is not used for making a decision, so just returns ABSTAIN
+ * @param authorizationCtx
+ * @param metadata
+ */
+export async function assignProjectInstanceId(
+  authorizationCtx: AuthorizationContext,
+  metadata: AuthorizationMetadata,
+) {
+  const projectId = authorizationCtx.invocationContext.args[0];
+  const resourceName = getResourceName(
+    metadata.resource ?? authorizationCtx.resource,
+    projectId,
+  );
+  // resourceId will override the resource name from metadata
+  authorizationCtx.resourceId = resourceName;
```

Comment:
Nitpick: would it be better to bind to the invocationContext instead of a property?

Reply:
@raymondfeng Hmm not sure what's the difference between "to the invocationContext" and "to a property", the property `resourceId` is a key(property) of `invocationContext` already..

Reply:
It’s a property on authorization context. I was referring to bind to invocation context.